nntrainer/src/neuralnet.cpp

```diff
@@ -527,6 +527,11 @@ int NeuralNetwork::init() {
       status = layers[i]->setOptimizer(opt);
       NN_RETURN_STATUS();
 
+      if (i == 0) {
+        ml_loge("Batch normalization layer cannot be first layer of network");
```

Comment:
Is this check needed?

Reply:
i could be 0 and i -1 is going to be negative.